### PR TITLE
Sponsors styled components

### PR DESCRIPTION
**Pull request by yarivShamash**

### CODE REVIEW

**Comment by yarivShamash** [comment]
Hey Serge,

I don't know why did it happen now but commits from previous PR are shown here, though I checked out to master and pulled before opening the current branch and PR.
So you can start your review with commit 180efdb .

File: client/src/components/SponsorsComps/HomePageSponsorsSection.jsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            .map((sponsor, i) => {
+              return (
+                <div key={i}>
+                  <SponsorMiniPremium key={sponsor._id} isOnWhite={true} {...sponsor} />
+                </div>
+                );
+          })}
+        </HomeSponsors>
+        <BreakLine /> 
+        <CommunitySponsorsSection> {/* NOTE: This is not shown when screen width is under 992px couldn't figure out why */}
```
Comment (sergelerner) [inline client/src/components/SponsorsComps/HomePageSponsorsSection.jsx]: Regarding the NOTE

Reversim_Summit_2019

File: client/src/components/SponsorsComps/HomePageSponsorsSection.jsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    align-items: baseline;
+    margin-bottom: ${space.xxl};
+  `}
+`;
+
+const Heading = styled.h2`
+  ${ ( { theme: { color, font } } ) => `
+        width: inherit;
+        position: relative;
+        z-index: 1;
```
Comment (sergelerner) [inline client/src/components/SponsorsComps/HomePageSponsorsSection.jsx]: I dont think we need 🤔

```
position: relative;
z-index: 1;
```

I removed it in developer and it looks fine 

Im Im not mistaken of course ... lets always aim for the minimum css needed

File: client/src/components/SponsorsComps/HomePageSponsorsSection.jsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        font-weight: ${font.weight_normal};
+        font-family: ${font.main};
+        font-size: ${font.size_h2};
+        color: ${color.heading_2};
+      `}
+`;
+
+const HomeSponsors = styled.div`
+  ${ ( { theme: { mq } } ) => `
+      max-width: 1200px;
```
Comment (sergelerner) [inline client/src/components/SponsorsComps/HomePageSponsorsSection.jsx]: I dont think we need it. When removed it is spread to 100% and is a little wider.

Anyway `SponserSectionContainter` the parent here, which acts as the Container, already has a max-width and some padding, so the children should just fall-in to that, make more sense.

File: client/src/components/SponsorsComps/HomePageSponsorsSection.jsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        <BreakLine /> 
+        <CommunitySponsorsSection> {/* NOTE: This is not shown when screen width is under 992px couldn't figure out why */}
+          <HomeCommunitySponsors sponsors={sponsors.filter(sponsor => !sponsor.isPremium)}/>
+        </CommunitySponsorsSection>
+      
+    </SponserSectionContainter>
+);
+};
+
+export default SponsorsSection;
```
Comment (sergelerner) [inline client/src/components/SponsorsComps/HomePageSponsorsSection.jsx]: For simplicity and sanity lets try to always have same export name as the file name, later when doing a search in the project it helps.

File: client/src/components/SponsorsComps/HomeCommunitySponsors.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
+/* eslint-disable prettier/prettier */
+import React from 'react';
+import { hyperlink } from "../../utils";
+import { image } from "../../images";
+import styled from "styled-components";
+
+// Styled-component Section
+
+const HomeCommunityContainer = styled.div`
```
Comment (sergelerner) [inline client/src/components/SponsorsComps/HomeCommunitySponsors.js]: OK, Important note: I am not doing this review chronologically 🙃meaning: I am going over every file from top to bottom, but I reviewed `HomePageSponsorsSection.jsx` first

So, suggested there "a different naming + breakdown" , there is a picture there I attached, and an example of jsx I wrote somewhere below - all in the sam file

Now, what I want to say here is that:


In `HomePageSponsorsSection.jsx`, there is 

```
<CommunitySponsorsSection>
          <HomeCommunitySponsors sponsors={sponsors.filter(sponsor => !sponsor.isPremium)}/></CommunitySponsorsSection>
```

Now, `CommunitySponsorsSection` has the following css

```
display: flex;
justify-content: space-between;
flex-wrap: wrap;
```

Now, in this file, you have a `HomeCommunityContainer`, which has same css

```
display: flex;
justify-content: space-between;
flex-wrap: wrap;
```

First of all, its redundant : )))

Second, `Section` and `Container` has a similar meaning in a way... Not necessarily, but lets have a consistent way of calling something that creates the **layout**  (flex stuf/ max-width...) for the component we include. `Container` works for me.

Third, Either we move the containers to the parent file, or have the container in each file, so it is self contained, better I think, usually.

File: client/src/components/SponsorsComps/HomeCommunitySponsors.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+  sponsorsOrder.forEach((key) => {
+    orderdSponsors.push(sponsors.find((sponsor) => sponsor.name === key))
+  })
+  return orderdSponsors
+}
+
+
+class SponsorMini extends React.Component {
+  state = {
+    hovered: false
```
Comment (sergelerner) [inline client/src/components/SponsorsComps/HomeCommunitySponsors.js]: I dont see this being used actually 

And if not, lets have a function component and not a class

File: client/src/components/SponsorsComps/HomeCommunitySponsors.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+  render() {
+    const { 
+      name,
+      logo,
+      url
+    } = this.props;
+    return (
+      <SponsorMiniLink href={hyperlink(url)} target="_blank">
+          <SponsorMiniImg
+            src={image(logo, 350, 230)}
```
Comment (sergelerner) [inline client/src/components/SponsorsComps/HomeCommunitySponsors.js]: nice! I didnt know we can do that :))) 

A styleComponent passes its props automatically to the Img....

File: client/src/components/SponsorsComps/HomeCommunitySponsors.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+  }
+}
+const HomeCommunitySponsors = ({sponsors}) => {
+  return (
+    <HomeCommunityContainer>
+    {orderSponsors(sponsors)
+      .map((sponsor, i) => {
+        return (
+          <SponsorMiniContainer key={i}>
+            <SponsorMini key={i} {...sponsor} />
```
Comment (sergelerner) [inline client/src/components/SponsorsComps/HomeCommunitySponsors.js]: When I think of it, we can kill `SponsorMini` and write it here... your call...

File: client/src/components/SponsorsComps/SponserGeneralComps.jsx [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+            </ImageContainer>
+            <SMPLinkingButton href={`/sponsor/${name}`}>
+              Explore Opportunities
+            </SMPLinkingButton>
+
+        </SponsorItem>
+      </div>
+    );
+  }
+};
```
Comment (sergelerner) [inline client/src/components/SponsorsComps/SponserGeneralComps.jsx]: I see there is a single export here.
Are e going to have more components in this `GeneralComps` ? 
If not, lets have a file per component, its easier to maintain.

File: client/src/components/SponsorsComps/SponserGeneralComps.jsx [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+    box-shadow: inset 0px 0px 10px 2px ${color.font_awsome_box_shadow_1}, 0px 0px 10px 0px ${color.font_awsome_box_shadow_2};
+  }
+`}
+`;
+
+//React components section
+
+export class SponsorMiniPremium extends React.Component {
+  state = {
+    hovered: false
```
Comment (sergelerner) [inline client/src/components/SponsorsComps/SponserGeneralComps.jsx]: not in use, can be a function component

File: client/src/components/SponsorsComps/SponserGeneralComps.jsx [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+  align-items: center;
+  margin: ${space.xl} ${space.s};
+`}
+`;
+
+const ImageContainer= styled.a`
+  ${( { theme: { color, space } } ) => `
+      width: 248px;
+      height: 230px;
+      max-width: 248px;
```
Comment (sergelerner) [inline client/src/components/SponsorsComps/SponserGeneralComps.jsx]: not sure we need both `max-width` and `width`

but maybe with images, there is some trick?

File: client/src/components/SponsorsComps/SponserGeneralComps.jsx [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+
+      position: relative;
```
Comment (sergelerner) [inline client/src/components/SponsorsComps/SponserGeneralComps.jsx]: not sure we need this either

File: client/src/components/SponsorsComps/SponserGeneralComps.jsx [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+      display: flex;
+      justify-content: center;
+      text-align: center;
```
Comment (sergelerner) [inline client/src/components/SponsorsComps/SponserGeneralComps.jsx]: same

File: client/src/components/SponsorsComps/SponserGeneralComps.jsx [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+
+const SponsorImg = styled.img`
+    ${ ({ theme: { space } }) => `
+      max-height: 240px;
+      max-width: 240px;
+      padding: ${space.s};
+    `}
+`;
+
+const SMPLinkingButton = styled(ButtonStyledLink)`
```
Comment (sergelerner) [inline client/src/components/SponsorsComps/SponserGeneralComps.jsx]: nice!

File: client/src/components/SponsorsComps/HomeCommunitySponsors.js (hunk 5 above)
Comment (yarivShamash) [reply client/src/components/SponsorsComps/HomeCommunitySponsors.js]: I'm on it.
Regarding naming, `Section`s are `<section>` and containers are `<div>`s.

File: client/src/components/SponsorsComps/HomeCommunitySponsors.js (hunk 8 above)
Comment (yarivShamash) [reply client/src/components/SponsorsComps/HomeCommunitySponsors.js]: I encountered some problems with this bit sent you a message in slack

File: client/src/components/SponsorsComps/SponserGeneralComps.jsx (hunk 9 above)
Comment (yarivShamash) [reply client/src/components/SponsorsComps/SponserGeneralComps.jsx]: I'm still not sure. Let's wait with it till the end of the sponsors bit.

**Pull request merged**